tests/playwright-test/web-server.spec.ts

```diff
@@ -744,3 +744,71 @@ test('should forward stdout when set to "pipe" before server is ready', async ({
   expect(result.output).toContain('[WebServer] output from server');
   expect(result.output).not.toContain('Timed out waiting 3000ms');
 });
+
+test.describe('kill option', () => {
+  test.skip(process.platform === 'win32', 'No sending SIGINT on Windows');
+
+  const files = (additionalOptions = {}) => {
+    const port = test.info().workerIndex * 2 + 10510;
+    return {
+      'web-server.js': `
+        process.on('SIGINT', () => { console.log('%%webserver received SIGINT but stubbornly refuses to wind down') })
+        process.on('SIGTERM', () => { console.log('%%webserver received SIGTERM but stubbornly refuses to wind down') })
+        const server = require('http').createServer((req, res) => { res.end("ok"); })
+        server.listen(process.argv[2], () => { console.log('webserver started'); });
+      `,
+      'test.spec.ts': `
+        import { test, expect } from '@playwright/test';
+        test('pass', async ({}) => {});
+      `,
+      'playwright.config.ts': `
+        module.exports = {
+          webServer: {
+            command: 'node web-server.js ${port}',
+            port: ${port},
+            stdout: 'pipe',
+            timeout: 3000,
+            ...${JSON.stringify(additionalOptions)}
+          },
+        };
+      `,
+    };
+  };
+
+  test('sends SIGKILL by default', async ({ interactWithTestRunner }) => {
+    const testProcess = await interactWithTestRunner(files(), { workers: 1 });
+
+    await testProcess.waitForOutput('webserver started');
+    process.kill(testProcess.process.pid!, 'SIGINT');
```

Comment:
I wonder why do you have to interact with the test runner? Shouldn't it send SIGINT/SIGTERM/SIGKILL during the normal testing operation without user interrupt? If so, I'd think that's more important to test.

Reply:
not sure why I did this. replaced it with bog-standard `runInlineTest`